### PR TITLE
Feature/calendar component

### PR DESCRIPTION
**Pull request by serjKim**
Implemented:
- Base calendar functionality: 
Components: pager, pages calendar, selecting days.
Drawing periods: dot, solid.
Support crossing periods.

- No swipe. Will be implemented probably next PR.

### CODE REVIEW

File: mobile/src/reducers/calendar/calendar-events.model.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import { Moment } from 'moment';
+
+export enum CalendarEventsType {
+    Vacation = 1,
+    SickLeave = 2,
+    Dayoff = 3,
+    AdditionalWork = 4
+}
+
+export enum CalendarEventStatus {
```
Comment (santee) [inline mobile/src/reducers/calendar/calendar-events.model.ts]: it's a string now

File: mobile/src/calendar/calendar-page.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    weekHeight: number;
+}
+
+export class CalendarPage extends Component<CalendarPageDefaultProps & CalendarPageProps, CalendarPageState> {
+    public static defaultProps: CalendarPageDefaultProps = {
+        hidePrevNextMonthDays: false
+    };
+
+    private readonly weekdaysNames = moment()
+        .locale('en')
```
Comment (demkalkov) [inline mobile/src/calendar/calendar-page.tsx]: Shouldn't we use device locale?

File: mobile/src/calendar/calendar-page-period.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import { calendarPeriodStyles, periodMargin } from './styles';
+import { View, StyleSheet } from 'react-native';
+
+const HalfPeriod = (props: {
```
Comment (santee) [inline mobile/src/calendar/calendar-page-period.tsx]: why lambda? why not a separate component?

File: mobile/src/calendar/calendar-page-period.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        <View style={containerStyles}>
+            <View style={circleStyles}></View>
+            <View style={periodStyles}></View>
+        </View>
+    );
+};
+
+type PeriodProps = { size: number, color: string };
+
+export const StartPeriod = (props: PeriodProps) => <HalfPeriod size={props.size} align={'left'} color={props.color} />;
```
Comment (santee) [inline mobile/src/calendar/calendar-page-period.tsx]: no, these can be lambdas, they are small :)

File: mobile/src/reducers/calendar/calendar.epics.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+            const endDate = moment(startDate);
+
+            endDate.add(14, 'days');
+
+            mockPeriod.startDate = startDate;
+            mockPeriod.endDate = endDate;
+            mockPeriod.startWorkingHour = 0;
+            mockPeriod.finishWorkingHour = 8;
+
+            mockEvent.calendarEventId = '1';
```
Comment (demkalkov) [inline mobile/src/reducers/calendar/calendar.epics.ts]: ~Why 1?~

File: mobile/src/reducers/calendar/calendar.epics.ts [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+            const startDate3 = moment();
+
+            startDate3.add(-9, 'days');
+
+            mockPeriod3.startDate = startDate3;
+            mockPeriod3.endDate = startDate3;
+            mockPeriod3.startWorkingHour = 0;
+            mockPeriod3.finishWorkingHour = 8;
+
+            mockEvent3.calendarEventId = '2';
```
Comment (demkalkov) [inline mobile/src/reducers/calendar/calendar.epics.ts]: `mockEvent3` with id = 2?

File: mobile/src/reducers/calendar/calendar.model.ts [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+    today: boolean;
+    belongsToCurrentMonth: boolean;
+}
+
+export interface WeekModel {
+    days: DayModel[];
+    weekIndex: number;
+}
+
+export class CalendarWeeksBuilder {
```
Comment (santee) [inline mobile/src/reducers/calendar/calendar.model.ts]: please move builders in separate files :)

File: mobile/src/calendar/styles.ts [hunk 8]
```diff
@@ -88,23 +171,21 @@ export const selectedDayStyles = StyleSheet.create({
         left: '50%',
         backgroundColor: '#fff',
         position: 'absolute',
-        transform: [{ translateX: -(circleDiameter / 2) }],
+        transform: [{ translateX: -50 }],
```
[the remark this answers is not in the document]
Comment (serjKim) [reply mobile/src/calendar/styles.ts]: lol) i don't know. It's a mistake

File: mobile/src/calendar/calendar-page.tsx (hunk 2 above)
Comment (serjKim) [reply mobile/src/calendar/calendar-page.tsx]: We sort of decided to use only 'en'. All design sketchs use en locale.

**Pull request merged**